framework/Optimizers/Optimizer.py

```diff
@@ -265,278 +203,78 @@ def _localWhatDoINeed(self):
-
-    # initialize dictionary entries
-    # TODO a bunch of the gradient-level trajectory initializations should be moved here.
-    for traj in self.optTraj:
-      self.optVars[traj]            = self.getOptVars()
-      self.submissionQueue[traj]    = deque()
-
-    #check initial point array consistency
-    rightLen = len(self.optTraj) #the hypothetical correct length
-    for var in self.getOptVars():
-      haveLen = len(self.optVarsInit['initial'][var])
-      if haveLen != rightLen:
-        self.raiseAnError(RuntimeError,'The number of trajectories for variable "{}" is incorrect!  Got {} but expected {}!  Check the <initial> block.'.format(var,haveLen,rightLen))
-
-    # check the constraint here to check if the initial values violate it
-    varK = {}
-    for trajInd in self.optTraj:
-      for varName in self.getOptVars():
-        varK[varName] = self.optVarsInit['initial'][varName][trajInd]
-      satisfied, _ = self.checkConstraint(varK)
-      if not satisfied:
-        # get a random value between the the lower and upper bounds
-        self.raiseAWarning("the initial values specified for trajectory "+str(trajInd)+" do not satify the contraints. Picking random ones!")
-        randomGuessesCnt = 0
-        while not satisfied and randomGuessesCnt < self.constraintHandlingPara['innerLoopLimit']:
-          for varName in self.getOptVars():
-            varK[varName] = self.optVarsInit['lowerBound'][varName]+randomUtils.random()*self.optVarsInit['ranges'][varName]
-            self.optVarsInit['initial'][varName][trajInd] = varK[varName]
-          satisfied, _ = self.checkConstraint(varK)
-        if not satisfied:
-          self.raiseAnError(Exception,"It was not possible to find any initial values that could satisfy the constraints for trajectory "+str(trajInd))
-
-    # extend multivalue variables (aka vector variables, or variables with "shape")
-    ## TODO someday take array of initial values from a DataSet
-    for var,shape in self.variableShapes.items():
-      if np.prod(shape) > 1:
-        for traj in self.optTraj:
-          baseVal = self.optVarsInit['initial'][var][traj]
-          newVal = np.ones(shape)*baseVal
-          self.optVarsInit['initial'][var][traj] = newVal
-
-    if self.initSeed is not None:
-      randomUtils.randomSeed(self.initSeed)
-
-    self.localInitialize(solutionExport=solutionExport)
+    AdaptiveSampler.initialize(self, externalSeeding=externalSeeding, solutionExport=solutionExport)
+    # functional constraints
+    for entry in self.assemblerDict.get('Constraint', []):
+      self._constraintFunctions.append(entry[3])
+    # sampler
+    self._initializeInitSampler(externalSeeding)
+    # seed
+    if self._seed is not None:
+      randomUtils.randomSeed(self._seed)
+    # variable bounds
+    self._variableBounds = {}
+    for var in self.toBeSampled:
+      dist = self.distDict[var]
```

Comment:
uniform dist only?

Reply:
I don't know of any reason we would be limited to uniform distributions. We only need upper and lower bounds, and infinity is acceptable.